### PR TITLE
Add astro65

### PR DESCRIPTION
**Pull request by swiftrax**
## Description
Add handwired/swiftrax/astro65

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by swiftrax** [comment]
image

File: keyboards/handwired/swiftrax/astro65/config.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x04D8
+#define PRODUCT_ID      0xEAEF
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Swiftrax
+#define PRODUCT         Astro65
+#define DESCRIPTION     65% keyboard with a daughterboard 
+
```
Comment (zvecr) [inline keyboards/handwired/swiftrax/astro65/config.h:29]:
```suggestion

```

File: keyboards/handwired/swiftrax/astro65/config.h [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+#define DIODE_DIRECTION COL2ROW
+
+/* define if matrix has ghost */
+//#define MATRIX_HAS_GHOST
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
+
+/*EEPROM for via*/
+#define DYNAMIC_KEYMAP_LAYER_COUNT 4
```
Comment (zvecr) [inline keyboards/handwired/swiftrax/astro65/config.h:49]: This is the default right now so having the define doesnt add much value
```suggestion
```

File: keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  KC_CAPS,   KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,   KC_SCLN,KC_QUOT,             KC_ENT,                 KC_PGUP,
+  KC_LSFT,   KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,                     KC_RSFT,       KC_UP,   KC_PGDN,
+  KC_LCTL,   KC_LGUI,KC_LALT,                KC_SPC,                        MO(1),   KC_RGUI,                           KC_LEFT, KC_DOWN, KC_RGHT),
+
+[1] = LAYOUT(
+  KC_GRV,    KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11,   KC_F12,   _______,  _______,      _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,  _______,  _______,                _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,            _______,                _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,                    _______,       _______, _______,
+  _______,   _______,_______,                _______,                       _______, _______,                           _______,_______,_______),
```
Comment (zvecr) [inline keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c:34]: Should add 2 more blank layers here

File: keyboards/handwired/swiftrax/astro65/rules.mk [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (zvecr) [inline keyboards/handwired/swiftrax/astro65/rules.mk:11]:
```suggestion
```

File: keyboards/handwired/swiftrax/astro65/rules.mk [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
Comment (zvecr) [inline keyboards/handwired/swiftrax/astro65/rules.mk:31]:
```suggestion
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/handwired/swiftrax/astro65/astro65.h [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include "quantum.h"
+
+// readability
+#define XXXX KC_NO
+
+#define LAYOUT( \
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/astro65.h:24]:
```suggestion
#define LAYOUT_65_ansi_blocker_split_bs( \
```

File: keyboards/handwired/swiftrax/astro65/info.json [hunk 7]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "Astro65", 
+    "url": "https://github.com/swiftrax", 
+    "maintainer": "Swiftrax", 
+    "width": 16, 
+    "height": 5, 
+    "layouts": {
+        "LAYOUT": {
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/info.json:8]:
```suggestion
        "LAYOUT_65_ansi_blocker_split_bs": {
```

File: keyboards/handwired/swiftrax/astro65/keymaps/default/keymap.c [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/keymaps/default/keymap.c:22]:
```suggestion
[0] = LAYOUT_65_ansi_blocker_split_bs(
```

File: keyboards/handwired/swiftrax/astro65/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  KC_ESC ,   KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,   KC_MINS,   KC_EQL,   KC_BSPC,   KC_BSLS,     KC_HOME,
+  KC_TAB ,   KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,   KC_LBRC,   KC_RBRC,  KC_BSLS,                KC_END ,
+  KC_CAPS,   KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,   KC_SCLN,KC_QUOT,             KC_ENT,                 KC_PGUP,
+  KC_LSFT,   KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,                     KC_RSFT,       KC_UP,   KC_PGDN,
+  KC_LCTL,   KC_LGUI,KC_LALT,                KC_SPC,                        MO(1),   KC_RGUI,                           KC_LEFT, KC_DOWN, KC_RGHT),
+
+[1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/keymaps/default/keymap.c:29]:
```suggestion
[1] = LAYOUT_65_ansi_blocker_split_bs(
```

File: keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include QMK_KEYBOARD_H
+
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+
+[0] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c:22]:
```suggestion
[0] = LAYOUT_65_ansi_blocker_split_bs(
```

File: keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+  KC_ESC ,   KC_1,   KC_2,   KC_3,   KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,   KC_0,   KC_MINS,   KC_EQL,   KC_BSPC,   KC_BSLS,     KC_HOME,
+  KC_TAB ,   KC_Q,   KC_W,   KC_E,   KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,   KC_P,   KC_LBRC,   KC_RBRC,  KC_BSLS,                KC_END ,
+  KC_CAPS,   KC_A,   KC_S,   KC_D,   KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,   KC_SCLN,KC_QUOT,             KC_ENT,                 KC_PGUP,
+  KC_LSFT,   KC_Z,   KC_X,   KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM,KC_DOT, KC_SLSH,                     KC_RSFT,       KC_UP,   KC_PGDN,
+  KC_LCTL,   KC_LGUI,KC_LALT,                KC_SPC,                        MO(1),   KC_RGUI,                           KC_LEFT, KC_DOWN, KC_RGHT),
+
+[1] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c:29]:
```suggestion
[1] = LAYOUT_65_ansi_blocker_split_bs(
```

File: keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+  KC_GRV,    KC_F1,  KC_F2,  KC_F3,  KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,  KC_F10, KC_F11,   KC_F12,   _______,  _______,      _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,  _______,  _______,                _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,            _______,                _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,                    _______,       _______, _______,
+  _______,   _______,_______,                _______,                        _______, _______,                           _______,_______,_______),
+
+[2] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c:36]:
```suggestion
[2] = LAYOUT_65_ansi_blocker_split_bs(
```

File: keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,  _______,  _______,  _______,      _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,  _______,  _______,                _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,_______,            _______,                _______,
+  _______,   _______,_______,_______,_______,_______,_______,_______,_______,_______,_______,                    _______,       _______, _______,
+  _______,   _______,_______,                _______,                        _______, _______,                           _______,_______,_______),
+
+[3] = LAYOUT(
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/keymaps/via/keymap.c:43]:
```suggestion
[3] = LAYOUT_65_ansi_blocker_split_bs(
```

File: keyboards/handwired/swiftrax/astro65/rules.mk [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (noroadsleft) [inline keyboards/handwired/swiftrax/astro65/rules.mk:22]:
```suggestion
AUDIO_ENABLE = no           # Audio output

LAYOUTS = 65_ansi_blocker_split_bs
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10449/files).

**Review by fauxpark — APPROVED** [review]

**Comment by noroadsleft** [comment]
https://github.com/qmk/qmk_firmware/pull/10449#discussion_r495756252 still needs updating.

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Comment by swiftrax** [comment]
Thanks!

**Pull request merged**